src/idom/core/layout.py

```diff
@@ -230,10 +230,17 @@ def _render_component(
         else:
             key, index = new_state.key, new_state.index
             parent.children_by_key[key] = new_state
-            # need to do insertion in case where old_state is None and we're appending
-            parent.model.current["children"][index : index + 1] = [
-                new_state.model.current
-            ]
+            # need to add this model to parent's children without mutating parent model
+            old_parent_model = parent.model.current
+            old_parent_children = old_parent_model["children"]
+            parent.model.current = {
+                **old_parent_model,
+                "children": [
+                    *old_parent_children[:index],
+                    new_state.model.current,
+                    *old_parent_children[index + 1 :],
+                ],
+            }
```

Comment:
Could this be the reason why VDOM children won't render if they follow a component?

Reply:
It might be related to the fact that I'm replacing the parent model on line 236? The fact that I'm not propagating this change up the model parents is probably _a problem_, but I can't quite see why that's causing your problem where the children are not rendering.